src/metrics/test_coverage/index.ts

```diff
@@ -0,0 +1,90 @@
+import { logger } from "../../core";
+import { decode } from "js-base64";
+import { keys, includes, find, propEq, toLower} from "ramda";
+import { DataEvent, MetricsSignature } from "../../interfaces";
+import {
+  WorkflowJobTestCoveragePayload,
+  WorkflowJobTestCoverageOutput,
+} from "./interfaces";
+import moment from "moment";
+import octokit from "../../core/octokit";
+import { isWorkflowJob } from "../../signatureConditions";
+
+export const collectWorkflowsTestCoverageMetrics = async (
+  dataEvent: DataEvent
+): Promise<DataEvent> => {
+  const payload = dataEvent.payload as WorkflowJobTestCoveragePayload;
+
+  const repo = payload.repository.name;
+  const owner = payload.repository.owner.login;
+  const completed_at = moment.utc(payload.workflow_job.completed_at).valueOf();
+  const created_at = moment.utc(payload.workflow_job.created_at).valueOf();
+  const started_at = moment.utc(payload.workflow_job.started_at).valueOf();
+  const duration = completed_at - started_at;
+  const status = payload.workflow_job.status;
+  const workflow_name = payload.workflow_job.workflow_name;
+  const steps = payload.workflow_job.steps.map((step) => ({   
+    ...step,
+    started_at: moment.utc(step.started_at).valueOf(),
+    completed_at: moment.utc(step.completed_at).valueOf(),
+    duration:
+      moment.utc(step.completed_at).valueOf() -
+      moment.utc(step.started_at).valueOf(),
+  }));
+
+  const isWorkflowNameTest = includes('test', toLower(workflow_name));
+  const stepRunTest = find(propEq('name', 'Run test'), steps);
```

Comment:
- [ ] Please, use a more generic rule. Only "Run test" isn't enough. You can just check is the word "test" si contained in the title.
- [ ] Maybe use just plain JS, not Ramda. I don't think that the checks work properly.